workbench-core/datasets/src/ddbDataSetMetadataPlugin.ts

```diff
@@ -122,7 +143,29 @@ export class DdbDataSetMetadataPlugin implements DataSetMetadataPlugin {
    * @param dataSetId - the ID of the Dataset to remove.
    */
   public async removeDataSet(dataSetId: string): Promise<void> {
-    await this._aws.helpers.ddb.delete(buildDynamoDBPkSk(dataSetId, this._dataSetKeyType)).execute();
+    const data = await this._aws.helpers.ddb.deleteItem({
+      key: buildDynamoDBPkSk(dataSetId, this._dataSetKeyType),
+      params: { return: 'ALL_OLD' }
+    });
+
+    const remainingDatasets = await this._storeStorageLocationToDdb(
+      {
+        name: data.storageName as string,
+        type: data.storageType as string,
+        awsAccountId: data.awsAccountId as string,
+        region: data.region as string
+      },
+      'decrement'
+    );
+
+    if (remainingDatasets < 1) {
```

Comment:
move this into _storeStorageLocationToDdb

Reply:
done